engine/session/git.go

```diff
@@ -233,26 +233,69 @@ func parseGitConfigOutput(output []byte) (*GitConfig, error) {
 
 	scanner := bufio.NewScanner(bytes.NewReader(output))
 
+	var currentKey string
+	var currentValue string
+	var inMultiline bool
+
 	for scanner.Scan() {
 		line := scanner.Text()
 		if line == "" {
 			continue
 		}
+
+		if inMultiline {
+			if strings.HasSuffix(line, "\\") {
+				currentValue += line[:len(line)-1] // remove trailing backslash
+			} else {
+				currentValue += line
+				inMultiline = false
+
+				if isGitConfigKeyAllowed(strings.ToLower(currentKey)) {
+					entries = append(entries, &GitConfigEntry{
+						Key:   currentKey,
+						Value: currentValue,
+					})
+				}
+
+				currentKey = ""
+				currentValue = ""
+			}
+			continue
+		}
+
 		parts := strings.SplitN(line, "=", 2)
 		if len(parts) != 2 {
 			return nil, fmt.Errorf("invalid format: line %q doesn't match key=value pattern", line)
 		}
 
-		if isGitConfigKeyAllowed(strings.ToLower(parts[0])) {
+		key := parts[0]
+		value := parts[1]
+
+		// check if this is the start of a multiline value
+		if strings.HasSuffix(value, "\\") {
+			inMultiline = true
+			currentKey = key
+			currentValue = value[:len(value)-1] // emove trailing backslash
+		} else if isGitConfigKeyAllowed(strings.ToLower(key)) {
+			entries = append(entries, &GitConfigEntry{
+				Key:   key,
+				Value: value,
+			})
+		}
```

Comment:
This feels duplicated from above. Can we refactor so that this block only appears once?

Reply:
not really the same block though, this one checks the start of the multiline while the above has is only valid while you're parsing a multi-line block and need to switch some variables. I could refactor but it won't improve readability that much since I still need to manage the special logic on each block :pray: